core/include/scipp/core/variable.h

```diff
@@ -446,6 +446,22 @@ class SCIPP_CORE_EXPORT Variable {
   units::Unit m_unit;
   VariableConceptHandle m_object;
 };
+/// This is used to provide the fabric function:
+/// createVariable<type>(Dims, Shape, Unit, Values<T1>, Variances<T2>)
+/// with the obligatory template argument type, function arguments are not
+/// obligatory and could be given in arbitrary order, but suposed to be wrapped
+/// in certain classes.
+/// Example: createVariable<float>(units::Unit(units::kg),
+/// Shape{1, 2}, Dims{Dim::X, Dim::Y}, Values({3, 4})).
+
+// The name should be changed to makeVariable after refactoring:
+// getting rid of all other makeVariable.
+template <class T, class... Ts> Variable createVariable(Ts &&... ts) {
```

Comment:
What is `T` for sparse data? With the existing `makeVariable` we specify the element type such as `double` when creating sparse data. Internally this then decides based on the dimensions (shape) whether it should be sparse data or not. We can also do this in a wrapper later.